Feature/documentation (#17)
EspeoBlockchain/gardener-server: docs/configuration.rst

-------------
Configuration
-------------

> Each part of gardener project (smart contracts repo, server, monitor) contains .env variables file.

Reviewer: ```suggestion
Each repository of Gardener project (smart contracts, server, monitor) contains an .env variables file.
```

> This section is going to explain parameters from them. Default parameters are set in a way everything should work correctly when using local test blockchain (ganache) and following getting-started guide.

Reviewer: ```suggestion
This section explains the available configuration parameters. Default parameters are set so that everything should work when using local test blockchain (ganache) and following the getting-started guide.
```
Link the getting started guide?

Server
======

- `PUBLIC_KEY` - address of the server's account, from which it sends the results of request to the smart contracts
- `PRIVATE_KEY` - private key of the server's account
- `ORACLE_ADDRESS` - address of the `Oracle` smart contract
- `DATABASE_URL` - URL with port for MongoDB connection
- `DATABASE_NAME` - MongoDB database name
- `NODE_URL` - URL for the provider to blockchain network

> - `API_PORT` - port for rest api request to server, currently only endpoint is heartbeat used by Gardener's monitor

Reviewer: ```suggestion
- `API_PORT` - port for exposing the server's REST API, currently the only endpoint is heartbeat used by Gardener's monitor
```

- `SAFE_DELAY_BLOCKS` - set a delay in a number of blocks to resist chain reorganization problem, server loads target block when it's number is at least `SAFE_DELAY_BLOCKS` lower than the youngest one
- `START_BLOCK` - set starting block number from which server listen for oracle requests

> - `PERSISTANCE` - selected persistance type, currently supported: `INMEMORY` or `MONGODB`

Reviewer: ```suggestion
- `PERSISTENCE` - selected persistence type, currently supported: `INMEMORY` or `MONGODB`
```
Wasn't this fixed in the server config?

Smart contracts
===============

- `PRIVATE_KEY` - private key used for contract deployment, if both `PRIVATE_KEY` and `MNEMONIC` are passed, `PRIVATE_KEY` is used
- `MNEMONIC` - 12 secret random words for accessing HD wallet (deployment)

> - `ROPSTEN_PROVIDER_URL` - provider for deploying contracts into ropsten test network

Reviewer: ```suggestion
- `ROPSTEN_PROVIDER_URL` - provider for deploying contracts into Ropsten test network
```
Why ROPSTEN_PROVIDER_URL and not just the PROVIDER_URL?

- `ORACLE_SERVER_ADDRESS` - address of oracle server account

Monitor
=======

> - `REACT_APP_NODE_URL` - URL for the provider to blockchain network

Reviewer: NODE_URL is inconsistent with the name PROVIDER_URL required for the smart contracts
